3_javascript/javascript3/week2/script.js

```diff
@@ -0,0 +1,75 @@
+
+
+// Function to fetch data from the API
+async function fetchData() {
+    try {
+        const response = await fetch('https://open.er-api.com/v6/latest/USD');
+        const data = await response.json();
+        return data;
+    } catch (error) {
+        console.error('Error fetching data:', error);
+        return null;
+    }
+}
+
+// Function to populate dropdowns with currency options
+function populateDropdowns() {
+    const fromCurrencyDropdown = document.getElementById('fromCurrency');
+    const toCurrencyDropdown = document.getElementById('toCurrency');
+
+    // Clear existing options
+    fromCurrencyDropdown.innerHTML = '';
+    toCurrencyDropdown.innerHTML = '';
+
+    // Check if currencyList is defined
+    if (currencyList) {
+        // Populate 'From' and 'To' dropdowns with currency options
+        currencyList.forEach(currency => {
+            const option1 = document.createElement('option');
+            option1.value = currency;
+            option1.textContent = currency;
+            fromCurrencyDropdown.appendChild(option1);
+
+            const option2 = document.createElement('option');
+            option2.value = currency;
+            option2.textContent = currency;
+            toCurrencyDropdown.appendChild(option2);
+        });
+
+        // Set default values for dropdowns
```

Comment:
The default values could have been assigned directly in the HTML.

Reply:
But here I couldn't do it. Maybe you can tell me what I'm doing wrong?

I made the following changes to the html:

`<label for="fromCurrency">From Currency:</label>
<select id="fromCurrency">
    <option value="USD">USD</option>
    <option value="EUR" selected>EUR</option>
</select>

<label for="toCurrency">To Currency:</label>
<select id="toCurrency">
    <option value="USD">USD</option>
    <option value="DKK" selected>DKK</option>
</select>`

But this only works if I comment out this line   <input type="number" id="amount" value="1" /> Please tell me what I don’t understand?

Reply:
```suggestion
        // Set default values for dropdowns
```
    <label for="amount">Amount:</label>
    <input type="number" id="amount" value="1" />

    <label for="fromCurrency">From Currency:</label>
    <select id="fromCurrency">
        <option value="USD">USD</option>
        <option value="EUR" selected>EUR</option>
    </select>
    
    <label for="toCurrency">To Currency:</label>
    <select id="toCurrency">
        <option value="USD">USD</option>
        <option value="DKK" selected>DKK</option>
    </select>

Reply:
I made these changes but they only work if I comment it out:
```suggestion
        // Set default values for dropdowns
```
    <input type="number" id="amount" value="1" />